### ISSUE TITLE
Empty link and missing header in Sheridan VAMC 

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#21404]
## Issue Description

Sheridan VAMC tugboat build has an empty blue block, complete with the empty link markup showing on all Facility pages and the System page. There’s an empty link teaser in the “Common links” of the Sheridan VAMC system node (seen at http://prod.cms.va.gov/node/6301).

<details>
<summary> Screenshots </summary>
![image](https://user-images.githubusercontent.com/55411834/110806348-36ddfb00-823f-11eb-84bc-8ca8e68b6c8a.png)

![image](https://user-images.githubusercontent.com/55411834/116295140-0225f480-a756-11eb-83b1-adc14604ebaf.png)
 

 ![image](https://user-images.githubusercontent.com/55411834/110806043-e8c8f780-823e-11eb-92cb-9bbea086bdda.png)

![image](https://user-images.githubusercontent.com/55411834/110806069-ef576f00-823e-11eb-9761-1fec28509e8c.png)
</details>

Improvements can be made in CMS to make it harder to introduce this bug in the content layer in the future.

## Rationale 
FE templates should always check for values before printing any wrapper HTML, even if the CMS is enforcing required fields. CMS doesn’t produce the actual <a> tags but may produce an empty array. 
- Sometimes required fields become optional
- Sometimes data is migrated in, like Facility data, which skirts form validation (required fields).

---
## Tasks
- [ ] Ensure FE templates checks for values before printing any wrapper HTML. 
- [ ] Add test to check 

## Acceptance Criteria
- [ ] If a CMS-managed content field is empty, the HTML wrapper should not be rendered.

**Comment by 1Copenut** [comment]
I'm seeing two pages on the Salt Lake City VAMC where we have no visual text. These are going to cause CI/CD breaks when the Nightwatch tests are fixed. Can we get these remediated as part of this ticket? Issue was confirmed on 4/8 at 12:15pm ET using axe-CLI.

* https://www.va.gov/salt-lake-city-health-care/locations/
* https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/

**Comment by 1Copenut** [comment]
@mmiddaugh The original URL in the ticket description ( http://prod.cms.va.gov/node/6301 ) redirects to an HTTPS view that doesn't look like the original screenshot when I view it with SOCKS. We're not sure how to get to this original view to fix the issue. Any guidance or suggestions?

/cc @timwright12 for visibility

**Comment by mmiddaugh** [comment]
@1Copenut have you tried accessing it from within the VA network?

**Comment by 1Copenut** [comment]
I just tried accessing it from the VA Network over VPN, with the same unexpected page shown.

**Comment by mpelzsherman** [comment]
I'm unclear about the scope of this ticket.

The "Relevant liquid templates" only pertain to News Stories.

Does this ticket pertain to News Stories only?

What is "wrapper HTML"?

**Comment by 1Copenut** [comment]
We have a case where there appears to be content coming from Drupal with empty `<a>` tags or a template that renders an empty `<a>` tag instead of nothing if there's not link text to fill it. This is throwing an error in axe because A tags cannot be on the page without visible or programmatically readable text.

**Comment by mmiddaugh** [comment]
It also occurred in the Common links box (not just in News Stories)

**Comment by mpelzsherman** [comment]
Unfortunately it is impossible to find all cases of this issue. Can we please limit the scope of this ticket to specic pages with known issues?

**Comment by 1Copenut** [comment]
Not really, no. This will get picked up in the daily axe scans when I ship my fix for Nightwatch issues failing silently, and it'll break CI builds. We will need to find all of them. I'll get a current-ish list later today or early Monday.

**Comment by mmiddaugh** [comment]
Based on the defects documented in this issue, it seems known pages are:

1. News stories
2. Locations -  [Trevor's comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21404#issuecomment-815956641) mentioned two sites with broken links - both are empty phone numbers - which I thought had already been resolved in #21791. It looks like [a fix is also pending on the CMS side](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/21404)
3. “Common links” node on system/facility pages - ie: Sheridan VAMC system node (seen at http://prod.cms.va.gov/node/6301))

**Comment by mpelzsherman** [comment]
There are literally hundreds of places in our templates where we render required content from the CMS without checking for it first.

Can we have a conversation about whether the FE needs to check for required CMS fields before rendering them?

**Comment by mpelzsherman** [comment]
Putting this ticket on hold until we get the list of axe scan issues from Trevor. Will likely turn into an epic if we need to wrap all links,` img, a`, etc.. on all pages.

**Comment by mpelzsherman** [comment]
Based on this [Slack comment](https://dsva.slack.com/archives/C01512KN35G/p1619472963441900?thread_ts=1619455024.430800&cid=C01512KN35G) by @jefflbrauer, it sounds like we should close this ticket?

**Comment by timwright12** [comment]
I opened a PR to deal with the two templates @1Copenut listed https://github.com/department-of-veterans-affairs/vets-website/pull/16976

I also found 16 instances in 12 files of `tel:{{` - someone them already had conditionals wrapping them, so if we want to tackle that it doesn't seem like a very big lift

**Comment by mmiddaugh** [comment]
Ok - this issue documents several completely separate issue so I will attempt to summarize here: 

**Empty link and missing header in Sheridan VAMC** 
- the empty link teaser was resolved by a content editor
- the missing header was a content issue, complicated by it being a non-required field. Kevin Walsh populated the field to resolve the immediate issue but we need to make a decision about how to address missing headers/default value going forward. 

**Missing visual text problem** 
- [Trevor's comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21404#issuecomment-815956641) above references missing phone numbers which have been addressed by @timwright12's [PR](department-of-veterans-affairs/vets-website#16976)
- The Facilities team is investigating this issue from the perspective of including archived/draft content in the build in #23481

**Comment by mmiddaugh** [comment]
@1Copenut this one can be validated on the following pages: 

Empty phone links
- https://www.va.gov/salt-lake-city-health-care/locations/ 
- https://www.va.gov/salt-lake-city-health-care/locations/cache-valley-va-clinic/ 

Empty links in blue Common links box
- https://www.va.gov/sheridan-health-care/

**Comment by timwright12** [comment]
This is still hung up in PR, it was requested that we also get some unit tests in there. Can someone point me to the empty links in https://www.va.gov/sheridan-health-care/ ? I'm not able to find them

**Comment by 1Copenut** [comment]
@timwright12 I just looked at Sheridan Health Care in prod a second ago. That issue appears to be resolved. 0 axe violations thrown.

**Comment by timwright12** [comment]
@mmiddaugh @1Copenut just merged this, it should be included in today's prod deploy to confirm https://github.com/department-of-veterans-affairs/vets-website/pull/16976
